modifiers/allocation/parse_mpibind_output.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import re
+import argparse
+
+
+def parse_mpibind(input_file):
+
+    with open(input_file, "r") as f:
+        lines = f.readlines()
+
+    task_map = {}
+
+    for line in lines:
+
+        # Match lines beginning with "mpibind:"
+        mpibind_match = re.match(r"^mpibind:.*(?:\r?\n|$)", line)
+        if mpibind_match:
+
+            pattern = r"task\s+(\d+)\s+nths\s+([\d,]*)\s+gpus\s+([\d,]*)\s+cpus\s+([\d,]*-[\d,]*)"
+            match = re.search(pattern, line)
+            if match:
+                task = match.group(1)  # Task number
+                nths = match.group(2)  # Nths value
+                gpus = match.group(3)  # GPUs value
+                cpus = match.group(4)  # CPUs value
+
+                task_map["task " + task] = {
+                    "cpus": cpus,
+                    "gpus": gpus,
+                    "nths": nths,
+                }
+
+    output_file = "mpibind_log_file.json"
+    with open(output_file, "w") as json_file:
```

Comment:
This looks relative: are you sure it runs with the basedir you want? You might want to specify an abspath (e.g. as another argument to `parse_mpibind_output`.